aws/resource_aws_workspaces_directory_test.go

```diff
@@ -216,6 +216,47 @@ func TestAccAwsWorkspacesDirectory_tags(t *testing.T) {
 	})
 }
 
+func TestAccAwsWorkspacesDirectory_ipGroupIds(t *testing.T) {
+	var v workspaces.WorkspaceDirectory
+	rName := acctest.RandString(8)
+
+	resourceName := "aws_workspaces_directory.main"
+	// resourceWorkspacesIpGroupName1 := "aws_workspaces_ip_group.test_alpha"
+	// resourceWorkspacesIpGroupName2 := "aws_workspaces_ip_group.test_omega"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckHasIAMRole(t, "workspaces_DefaultRole") },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsWorkspacesDirectoryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccWorkspacesDirectoryConfig_ipGroupIds_create(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsWorkspacesDirectoryExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ip_group_ids.#", "1"),
+					// CONSIDER: For some reason returns error: "Attribute "ip_group_ids.0" not set"
```

Comment:
Have no idea why test check fails here 🤷‍♂️

Reply:
it fails because because `ip_group_ids` is a set and not an ordered list. the "index" will a hash value instead. you can leave as just the count test and the import will test the rest.

Reply:
Ah, I see. Does it make sense to switch from `Set` to `List` for this attribute? I remember @gdavison suggested to use `TypeList + MaxItem 1` over `TypeSet + MaxItem 1`, but what is the best for many items?

Reply:
How does `import` know what should be tested? Does `ImportStateVerify` compare resource configuration attributes and actual state attributes?

Reply:
yep, let say that if would have omitted the read part for the attribute the import test would have showed the diff.

Reply:
Thank you 🙇‍♂️